docs/sample_files/mailhog/docker-compose.sample.yml

```
@@ -0,0 +1,26 @@
+version: '3.8'
+services:
+  mailhog:
+    image: jcalonso/mailhog:latest
+    restart: unless-stopped
+    environment:
+      VIRTUAL_HOST: mailhog.dev.test
+      VIRTUAL_PORT: 8025
+    ports:
+      - 1025:1025 # SMTP
+      - 8025:8025 # web
+    labels:
+      - traefik.enable=true
+      - traefik.http.routers.mailhog.entrypoints=web
+      - traefik.http.routers.mailhog.rule=Host(`mailhog.dev.test`)
+      - traefik.http.services.mailhog_https.loadbalancer.server.port=8025
+      - traefik.http.routers.mailhog_https.rule=Host(`mailhog.dev.test`)
+      - traefik.http.routers.mailhog_https.tls=true
+      - traefik.http.routers.mailhog_https.entrypoints=web-secure
+      - traefik.http.middlewares.mailhog_https.redirectscheme.scheme=https
+      - traefik.http.routers.mailhog.middlewares=mailhog_https
```

It may be nice to clean these up a bit more and maybe have the https as the "default" rule by name (i.e. mailhog instead of mailhog_https) and then add on the http label as it's only there for redirects. (Note: this has not been tested)

```suggestion
      # Traefik: enable integration and map internal service port
      - traefik.enable=true
      - traefik.http.services.mailhog.loadbalancer.server.port=8025

      # HTTPS router (served on web-secure)
      - traefik.http.routers.mailhog.rule=Host(`mailhog.dev.test`)
      - traefik.http.routers.mailhog.entrypoints=web-secure
      - traefik.http.routers.mailhog.tls=true

      # HTTP -> HTTPS redirect middleware
      - traefik.http.middlewares.redirect-to-https.redirectscheme.scheme=https

      # HTTP router (web) that only redirects to HTTPS
      - traefik.http.routers.mailhog-redirect.rule=Host(`mailhog.dev.test`)
      - traefik.http.routers.mailhog-redirect.entrypoints=web
      - traefik.http.routers.mailhog-redirect.middlewares=redirect-to-https
```